packages/components/plh/parent-group/plh-parent-group.service.ts

```diff
@@ -547,6 +571,59 @@ export class PlhParentGroupService extends SyncServiceBase {
     );
   }
 
+  private isParentFromRapidPro(
+    parent: IParent | IParentFromRapidPro
+  ): parent is IParentFromRapidPro {
+    return (parent as IParentFromRapidPro).rapidpro_uuid !== undefined;
+  }
+
+  /**
+   * Convert parent data added to shared data from RapidPro into local parent data format
+   *
+   * Example input:
+   * ```
+   * {
+   *   rapidpro_uuid: "uuid-123",
+   *   rapidpro_fields: {
+   *     name: "Jasper",
+   *     age: 10,
+   *     custom_field: "value",
+   *   },
+   * }
+   * ```
+   *
+   * Example output:
+   * ```
+   * {
+   *   id: "uuid-123",
+   *   group_id: "group-456",
+   *   rapidpro_uuid: "uuid-123",
+   *   rp_name: "Jasper",
+   *   rp_age: 10,
+   *   rp_custom_field: "value",
+   * }
+   * ```
+   *
+   * @param parent - Parent data from RapidPro
+   * @param parentGroupId - ID of the parent group
+   * @returns Formatted parent data
+   */
+  private hackFormatParentFromRapidPro(
+    parent: IParentFromRapidPro,
+    parentGroupId: string
+  ): IParent {
+    const { rapidpro_uuid, rapidpro_fields, ...rest } = parent;
+    const parsedRapidProFields = Object.fromEntries(
+      Object.entries(rapidpro_fields).map(([key, value]) => [`rp_${key}`, value])
+    );
+    return {
+      ...rest,
+      ...parsedRapidProFields,
+      id: rapidpro_uuid,
+      group_id: parentGroupId,
+      rapidpro_uuid,
```

Comment:
nit(non-blocking): you might want to just keep as `rp_uuid` to save maintaining 2 separate prefixes related to rapidpro. I feel like there's probably not a massive need to distinguish data coming from rapidpro fields vs rapidpro meta give the only non-field is the uuid.